FeedStoreChallenge/ManagedCache.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ManagedCache.swift
+//  FeedStoreChallenge
+//
+//  Created by Joshua Bryson on 9/9/21.
+//  Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import CoreData
+
+@objc(ManagedCache)
+internal class ManagedCache: NSManagedObject {
+	@NSManaged internal var timestamp: Date
+	@NSManaged internal var feed: NSOrderedSet
+}
+
+extension ManagedCache {
+	static func deletePreviousCache(in context: NSManagedObjectContext) throws {
+		try find(in: context).map(context.delete).map(context.save)
+	}
+
+	static func newUniqueInstance(in context: NSManagedObjectContext) throws -> ManagedCache {
+		try deletePreviousCache(in: context)
+		return NSEntityDescription.insertNewObject(forEntityName: String(describing: ManagedCache.self), into: context) as! ManagedCache
```

Comment:
Suggestion:
Entity name and class name potentially can be different. Thus it is better to use `entity().name!`
```swift
return NSEntityDescription.insertNewObject(forEntityName: entity().name!, into: context) as! ManagedCache
```
And the whole return can be simplified to the following
```suggestion
		return ManagedCache(context: context)
```